### DISCUSSION TITLE
Checkout can't be completed when a line item with a selling plan is added to the cart

### DISCUSSION THREAD

**Discussion by abuiles** [Shopify/function-examples#220] [Feedback: Cart Transform API]
Given product A and B that can be merged into product C, where A and B can be bought as subscription. 

1. Add product A and B to the cart, each with a selling plan.
2. Call the `MergeOperation` with the following parameters:
```
MergeOperation {
  parentVariantId: C.variantId
  cartLines: [A, B]
}
```
3. Go to `/checkout` and try to pay.

### Expected behavior:

The checkout process should be completed successfully, even if there are line items in the cart with a selling plan. Each bundle item should keep the selling plan after checkout and create a subscription contract.

### Actual behavior:

The checkout process cannot be completed when a line item with a selling plan is added to the cart.

Checkout response: 

```
{
  "errors": [
    {
      "code": "MERCHANDISE_EXPECTED_PRICE_MISMATCH",
      "localizedMessage": "",
      "nonLocalizedMessage": "The price of this line could not be honoured, new price reflected in counter proposal.",
      "localizedMessageHtml": null,
      "message": {
        "code": "MERCHANDISE_EXPECTED_PRICE_MISMATCH",
        "localizedDescription": null,
        "__typename": "ViolationErrorMessage"
      },
      "from": "$[0]",
      "to": "$[0]",
      "__typename": "ConfirmChangeViolation"
    },
    {
      "code": "PAYMENTS_SUBSCRIPTIONS_TERMS_NOT_ACCEPTED",
      "localizedMessage": "",
      "nonLocalizedMessage": "Subscription terms not accepted.",
      "localizedMessageHtml": null,
      "message": {
        "code": "PAYMENTS_SUBSCRIPTIONS_TERMS_NOT_ACCEPTED",
        "localizedDescription": null,
        "__typename": "ViolationErrorMessage"
      },
      "target": "$.payment",
      "__typename": "UnprocessableTermViolation"
    }
  ]
}
```

![checkout](https://user-images.githubusercontent.com/21772/233697221-9f3ab510-d99f-4864-917e-5307da288e93.gif)

**Comment by ignacio-chiazzo** [comment]
Hey! 

The case you're trying is Subscription at the component level. Could you let me know if you're still getting that error?


We aren’t currently supporting `Selling Plans` with bundles, but we have plans to enable compatibility later this year. We’ll keep sending an update as soon as the capability is available.

We have two cases:

- **Subscriptions at the Bundle level**: e.g. A _Monthly Kit_ subscription.
- **Subscriptions at the Component level**: e.g. A _Coffee Bundle kit_ that contains _1 Coffee Machine_ and a _12 month coffee subscription_.

We have two operations, `merge` and `expand`. The current scenarios are:

**Expand operation:**
- Subscription at the parent level: A selling plan could be assigned to the parent product and then expand the items. 
- Subscription at the component level: After expanding the item, a selling plan is assigned to a component.

**Merge operation:**
- Subscription at the parent level: after merging the items, a selling plan is assigned to the parent.
- Subscription at the component level: A selling plan is assigned to the items to be merged.

**Reply by sadsciencee** [reply]
hey @ignacio-chiazzo - any chance you're able to provide an eta for compatibility? we have some plus merchants asking for a timeline as this is a blocker to migrate off ruby scripts

**Reply by anthony-v-sundae-lane** [reply]
hey @ignacio-chiazzo  Just checking in on the progress of this one. Is it looking like a candidate for any of this year's API releases?

**Reply by AdamHoK** [reply]
Hi @ignacio-chiazzo We would like to know if this is still in the pipeline, and when we should be able to expect this functionality

**Comment by ProgramYourFace** [comment]
Any updates on this? This functionality would greatly reduce the complexity of our frontend and backend cart logic and make migrating away from checkout.liquid so much easier.

**Comment by ctrevorrow** [comment]
@ignacio-chiazzo  Could you please provide any updates as to when selling plan support might be coming to Cart Transforms?  Or if that is still planned?
Last June, you said there were plans to enable compatibility later in 2023, so it seems the timeline was pushed out or the plans were scrapped completely.  
My Plus merchant requires selling plan support for bundles, and I'd like to update them on when/if that will be possible as I previously said it was "planned for Q4'23".

**Comment by artyrcheek** [comment]
Any updates on this?

**Comment by joshistoast** [comment]
Checking in to see if there's any updates from Shopify on this 👀

**Comment by ejprismfly** [comment]
Almost all of the big shopify merchants now a days are using bundles and wants a way to properly display it on all pages together with a subscription or discounts. Hopefully we could cater this instead of going hacky again.

**Comment by nghia8624** [comment]
Is it updated?

**Comment by anthony-v-sundae-lane** [comment]
@ignacio-chiazzo or @nickwesselman - This thread has turned into a feature request, but I'd like to check if the actual bug reported will be fixed.

If a bundle is added to cart that contains a subscription-only product as a component item, Cart Transform Function automatically makes itself valid by removing the subscription product from the bundle.

This allows the bundle to be added to cart, and for the customer to proceed to checkout.

**Shopify payments then infinitely fails on every request with no errors returned - the user cannot checkout.**

This is a pretty massive bug. Could we get an update or resolution on this?

**Comment by BrennanTanner** [comment]
Bumping this, we would really like to see this functionality implemented, or at the very least be given a final answer on whether it is being implemented ever.
